src/workers/atlas_resolvers/atlas_transfer_resolver.js

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright: Ambrosus Inc.
+Email: tech@ambrosus.com
+
+This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
+*/
+
+import promClient from 'prom-client';
+import AtlasParticipationStrategy from '../atlas_strategies/atlas_participation_strategy';
+import {AtlasResolver, atlasResolutionStatus} from './atlas_resolver';
+import {getDefaultAddress} from '../../utils/web3_tools';
+
+export default class AtlaTransferResolver extends AtlasResolver {
+  constructor(
+    web3,
+    dataModelEngine,
+    transfersRepository,
+    failedTransfersCache,
+    strategy,
+    workerLogRepository,
+    logger
+  ) {
+    super();
+    this.web3 = web3;
+    this.dataModelEngine = dataModelEngine;
+    this.strategy = strategy;
+    this.transfersRepository = transfersRepository;
+    this.failedTransfersCache = failedTransfersCache;
+    this.workerLogRepository = workerLogRepository;
+    this.logger = logger;
+
+    if (!(this.strategy instanceof AtlasParticipationStrategy)) {
+      throw new Error('A valid transfer strategy must be provided');
+    }
+  }
+
+  async tryToResolve({bundleId}, {transferId}) {
+    await this.transfersRepository.resolveTransfer(transferId);
+    await this.dataModelEngine.markBundleAsSheltered(bundleId);
+    await this.addLog('🍾 Yahoo! The bundle is ours.', {bundleId});
+  }
+
+  async tryToDownload({donorId, bundleId, transferId}) {
+    const transferExpirationTime = await this.transfersRepository.getTransferExpirationTimeInMs(transferId);
```

Comment:
The method `getTransferExpirationTimeInMs` does not exist in `transfersRepository`.